tst/standard/cong.tst

```diff
@@ -324,10 +324,98 @@ gap> cong2 := SemigroupCongruence(S, pair2);;
 gap> IsSuperrelation(cong1, cong2);
 true
 
+#T# Equality for different types of congruence class
+gap> S := FullTransformationMonoid(4);;
+gap> I := SemigroupIdeal(S, Transformation([1,1,2,2]));;
+gap> reescong := ReesCongruenceOfSemigroupIdeal(I);;
+gap> pair := [Transformation([1, 1, 2, 2]), Transformation([1, 1, 1, 1])];;
+gap> cong := SemigroupCongruence(S, pair);;
+gap> reesclass := EquivalenceClassOfElement(reescong, pair[1]);
+<congruence class of Transformation( [ 1, 1, 2, 2 ] )>
+gap> class := EquivalenceClassOfElement(cong, Transformation([1,1,2,2]));
+<congruence class of Transformation( [ 1, 1, 2, 2 ] )>
+gap> class = reesclass;
+true
+gap> cong := SemigroupCongruence(S, []);;
+gap> class := EquivalenceClassOfElement(cong, Transformation([1,1,2,2]));;
+gap> class = reesclass;
+false
+gap> cong := UniversalSemigroupCongruence(S);;
+gap> class := EquivalenceClassOfElement(cong, Transformation([1,1,2,2]));;
+gap> class = reesclass;
+false
+
+#T# EquivalenceRelation(Canonical)Lookup
+gap> S := FullTransformationMonoid(3);;
+gap> I := SemigroupIdeal(S, Transformation([1, 1, 2]));;
+gap> cong := ReesCongruenceOfSemigroupIdeal(I);;
+gap> EquivalenceRelationLookup(cong);
+[ 1, 2, 3, 11, 5, 6, 11, 8, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 
+  11, 11, 11, 11, 11, 11, 11 ]
+gap> EquivalenceRelationCanonicalLookup(cong);
+[ 1, 2, 3, 4, 5, 6, 4, 7, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 
+  4, 4 ]
+
+# EquivalenceRelationCanonicalLookup with an RMS cong
+gap> S := ReesMatrixSemigroup(SymmetricGroup(3), [[(1,2), ()], [(), (1,3)]]);
+<Rees matrix semigroup 2x2 over Sym( [ 1 .. 3 ] )>
+gap> cong := RMSCongruenceByLinkedTriple(S, Group((1,2,3)),
+>                                        [[1],[2]], [[1,2]]);;
+gap> ccong := AsSemigroupCongruenceByGeneratingPairs(cong);;
+gap> cong := RMSCongruenceByLinkedTriple(S, Group((1,2,3)),
+>                                        [[1],[2]], [[1,2]]);;
+gap> EquivalenceRelationCanonicalLookup(cong);
+[ 1, 2, 3, 1, 2, 2, 2, 1, 1, 3, 4, 4, 2, 1, 1, 2, 3, 4, 4, 4, 3, 3, 3, 4 ]
+gap> EquivalenceRelationCanonicalLookup(ccong);
+[ 1, 2, 3, 1, 2, 2, 2, 1, 1, 3, 4, 4, 2, 1, 1, 2, 3, 4, 4, 4, 3, 3, 3, 4 ]
+gap> cong = ccong;
+true
```

Comment:
It should be the lookup one - as above, we reconstruct `cong` to make sure it doesn't have generating pairs.